processors/minsev/minsev_test.go

```diff
@@ -0,0 +1,191 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package minsev
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	api "go.opentelemetry.io/otel/log"
+	"go.opentelemetry.io/otel/sdk/log"
+)
+
+var severities = []api.Severity{
+	api.SeverityTrace, api.SeverityTrace1, api.SeverityTrace2, api.SeverityTrace3, api.SeverityTrace4,
+	api.SeverityDebug, api.SeverityDebug1, api.SeverityDebug2, api.SeverityDebug3, api.SeverityDebug4,
+	api.SeverityInfo, api.SeverityInfo1, api.SeverityInfo2, api.SeverityInfo3, api.SeverityInfo4,
+	api.SeverityWarn, api.SeverityWarn1, api.SeverityWarn2, api.SeverityWarn3, api.SeverityWarn4,
+	api.SeverityError, api.SeverityError1, api.SeverityError2, api.SeverityError3, api.SeverityError4,
+	api.SeverityFatal, api.SeverityFatal1, api.SeverityFatal2, api.SeverityFatal3, api.SeverityFatal4,
+}
+
+type args struct {
+	Ctx    context.Context
+	Record log.Record
+}
+
+type processor struct {
+	ReturnErr error
+
+	OnEmitCalls     []args
+	EnabledCalls    []args
+	ForceFlushCalls []context.Context
+	ShutdownCalls   []context.Context
+}
+
+func (p *processor) OnEmit(ctx context.Context, r log.Record) error {
+	p.OnEmitCalls = append(p.OnEmitCalls, args{ctx, r})
+	return p.ReturnErr
+}
+
+func (p *processor) Enabled(ctx context.Context, r log.Record) bool {
+	p.EnabledCalls = append(p.EnabledCalls, args{ctx, r})
+	return true
+}
+
+func (p *processor) Shutdown(ctx context.Context) error {
+	p.ShutdownCalls = append(p.ShutdownCalls, ctx)
+	return p.ReturnErr
+}
+
+func (p *processor) ForceFlush(ctx context.Context) error {
+	p.ForceFlushCalls = append(p.ForceFlushCalls, ctx)
+	return p.ReturnErr
+}
+
+func (p *processor) Reset() {
+	p.OnEmitCalls = p.OnEmitCalls[:0]
+	p.EnabledCalls = p.EnabledCalls[:0]
+	p.ShutdownCalls = p.ShutdownCalls[:0]
+	p.ForceFlushCalls = p.ForceFlushCalls[:0]
+}
+
+func TestLogProcessorOnEmit(t *testing.T) {
+	t.Run("Passthrough", func(t *testing.T) {
+		wrapped := &processor{ReturnErr: assert.AnError}
+
+		p := NewLogProcessor(wrapped, api.SeverityTrace1)
+		ctx := context.Background()
+		r := &log.Record{}
```

Comment:
If we would move this inside the `for` loop here and in other table-driven-tests then we would not need to use (and define) `wrapped.Reset()`, right?

Reply:
No, the `OnEmitCalls` slice would still be appended to and need to be reset.